tools/backups2datalad/datasetter.py

```diff
@@ -557,3 +559,72 @@ async def debug_logfile(self, quiet_debug: bool) -> None:
         handler.setFormatter(fmter)
         root.addHandler(handler)
         log.info("Saving logs to %s", self.logfile)
+
+    async def get_superds_commit_message(self, superds: AsyncDataset) -> str:
+        added = []
+        modified = []
+        other = False
+        for entry in split_terminated(
+            await superds.read_git(
+                "status", "--porcelain", "--ignore-submodules=dirty", "-z", strip=False
+            ),
+            "\0",
+        ):
+            status = entry[:2]
+            path = entry[3:].rstrip("/")
+            if status[0] == "A" or status == "??":
+                added.append(path)
+            elif status[0] == "M" or status[1] == "M":
+                modified.append(path)
+            else:
+                other = True
+        added_datasets: dict[str, list[str]] = {}
+        modified_datasets: dict[str, list[str]] = {}
+        for path in added:
+            if (superds.pathobj / path / ".git").exists():
+                commits = (
+                    await AsyncDataset(superds.pathobj / path).read_git(
+                        "log", "--format=%s"
+                    )
+                ).splitlines()
+                added_datasets[path] = commits
+            else:
+                other = True
+        for path in modified:
+            if (superds.pathobj / path / ".git").exists():
+                last_commit = (
+                    (await superds.read_git("ls-tree", "-z", "HEAD", "--", path))
+                    .partition("\t")[0]
+                    .split()[2]
+                )
+                commits = (
+                    await AsyncDataset(superds.pathobj / path).read_git(
+                        "log", "--format=%s", f"{last_commit}.."
+                    )
+                ).splitlines()
+                modified_datasets[path] = commits
+            else:
+                other = True
+        msg = ""
+        if added_datasets:
+            msg += f"{len(added_datasets)} added"
+            if len(added_datasets) <= 3:
+                msg += " (" + ", ".join(sorted(added_datasets.keys())) + ")"
```

Comment:
shorter and IMHO better

```suggestion
                msg += " (", ", ".join(sorted(added_datasets.keys())) + ")"
```

and the same in duplicated code below (or better come up/use helper or some loop to centralize logic/avoid duplication)

Reply:
also please add `else` statement where instead of listing, have `{len(added_datasets)} datasets` to provide summary.

Reply:
* Your suggested change would mean that you're adding a tuple to string, which doesn't work.  What are you trying to do?
* The number of datasets is already included in the `f"{len(added_datasets)} added"` string immediately above.

Reply:
> * Your suggested change would mean that you're adding a tuple to string, which doesn't work.  What are you trying to do?

I misread the code !  Thank you, let's consider this one resolved